pkg/generator/generate_inputs.go

```diff
@@ -16,11 +16,13 @@ type GenerateInputs struct {
 }
 
 func (i GenerateInputs) Add(anns map[string]string) {
-	bs, err := json.Marshal(i.inputs)
-	if err != nil {
-		panic("Cannot marshal generate inputs")
+	if len(anns) > 0 {
+		bs, err := json.Marshal(i.inputs)
+		if err != nil {
+			panic("Cannot marshal generate inputs")
+		}
+		anns[GenerateInputsAnnKey] = string(bs)
```

Comment:
@vicmarbev - First: Thank you for contributing!

I agree that this will fix the crash but I'm honestly not sure whether this will lead to the desired behavior, as I haven't spent enough time understanding this part of the controller at all yet.

The way this is written, I think we won't add the "generate-inputs" annotation in some cases, and my suspicion is that this will lead to incorrect behavior or "surprising UX" someplace else.

  Do you think we should always add our "secretgen.k14s.io/generate-inputs" annotation even if the secret had no other previous annotations?

Reply:
Sure! What should be the value for  `secretgen.k14s.io/generate-inputs` if no annotations are provided?

Reply:
@vicmarbev It looks like one way out of this is to try to ensure that the annotations on the secret aren't nil when we call this `Add` function: https://github.com/vmware-tanzu/carvel-secretgen-controller/blob/develop/pkg/generator/certificate_reconciler.go#L119

(1)
Above that line we could check whether the Secret object has nil annotations and then instantiate an empty map into its Annotations field.

(2)
Additionally it would be great to change this panic to be more informative that the caller is calling the function in an unanticipated way. For example we talked with @joaopapereira about what imgpkg does, and it might be more informative to have an error message saying something like, "Internal inconsistency: <function name> called with nil param". 

(3)
Finally, to be the most cloud-native-y and live our best lives, we should probably convert this from a panic into a "comma, error" idiom and have the reconciler (linked above) mark the resource as "failed to reconcile" if the `Add` function can't execute. 


For the scope of this PR please definitely do (1) above.  In terms of making future failures more graceful, we'd prefer to skip (2) and go straight to (3), but given that this code is really only currently called in one place, if you want to resubmit with just (1) implemented that will be an incremental improvement.

re. your question about the appropriate value of the annotation - honestly it looks like this part of the code isn't fully finished since nobody ever finished the TODO in the `UpdateSecret` method.  The current `bs` value is fine.

cc @neil-hickey @joaopapereira